XenCenterLib/StringUtility.cs

```diff
@@ -218,5 +218,21 @@ public static bool IsIPAddress(string s)
 
             return true;
         }
+
+        /// <summary>
+        /// To be used to format file paths for File streams. <br />
+        /// Prepends \\?\ to path if it is longer than the default MAX_PATH in Windows. <br /><br />
+        /// See <see href='https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea'>CreateFileA</see> and <see href='https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry'>Maximum Path Length Limitation</see> for more info.
+        /// </summary>
+        /// <param name="inputPath">The input file path</param>
+        /// <returns></returns>
+        public static string FormatFilePathForFileStream(string inputPath)
+        {
+            if (string.IsNullOrEmpty(inputPath) || inputPath.Length <= 260 || inputPath.StartsWith("\\\\?\\"))

```

Comment:
- Maybe rename this method to `ToLongPath" or something so it's more obvious what it does.
- Wouldn't it be easier on the eye if we used a verbatim string @""? :dizzy_face:
- I've been testing this and I believe it should be `inputPath.Length < 260` - the equality crashes for me (I think it's because of the `<nul>` end character). It's also giving me an additional restriction when I'm trying to create a directory to write a file: in `directory.Lengh < 248`, can we check whether need to take this into account too when adding directories.
- There are some unit tests around the ArchiveIterator/Writer; given the nature of the change, it may be a good idea to see if we can add some cases.

Reply:
+1 on name change